manifests/install.pp

```diff
@@ -124,15 +121,24 @@
       refreshonly => true,
     }
   } elsif $::operatingsystem == 'Darwin' and $::macosx_productversion_major =~ /^10\.(9|10|11|12|13)/ {
-    contain puppet_agent::install::remove_packages
-
-    # package provider does not provide 'versionable'
-    package { $::puppet_agent::package_name:
-      ensure          => 'present',
-      provider        => 'pkgdmg',
-      source          => "/opt/puppetlabs/packages/${package_file_name}",
-      require         => Class['puppet_agent::install::remove_packages'],
-      install_options => $install_options,
+    if $old_packages or $puppet_agent::aio_upgrade_required {
+      $install_script = "osx_install.sh.erb"
+  
+      contain puppet_agent::install::remove_packages
+  
```

Comment:
Dunno if it'd be useful to refactor this and the Sol 10. stuff into a defined type? Something like `perform_agent_upgrade { <platform> }`? I chose not to do it here since the code seems simple enough, and b/c our Sol 10 agents use ctrun to execute the script (here: https://github.com/puppetlabs/puppetlabs-puppet_agent/blob/master/manifests/install.pp#L99) -- I don't want to introduce a weird param. like $script_wrapper when this code is only used twice.. I don't know why ctrun is used, as from what I understand that code basically runs the script in the background. We don't use any of ctrun's event reporting features. Maybe that could be refactored to just execute the script in the background? Or we could use a Mac-OS' equivalent of ctrun? I couldn't find any examples of the latter.

Reply:
I don't think it's worth factoring out yet.

As for ctrun, I have no idea why we use that instead of just backgrounding the script.